### PR TITLE
Edit screen navigation.

### PR DESCRIPTION
**Pull request by kejmil01**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,6 +18,7 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".mvvm.edit.EditActivity"></activity>
```
Comment (BartSoftware) [inline app/src/main/AndroidManifest.xml]: closing tag is redundant

File: app/src/main/java/com/invisibleteam/goinvisible/model/ImageDetails.java [hunk 2]
```diff
@@ -16,4 +19,33 @@ public String getPath() {
...
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeString(path);
+        dest.writeString(name);
+    }
+
+    @SuppressWarnings("unused")
+    public static final Parcelable.Creator<ImageDetails> CREATOR = new Parcelable.Creator<ImageDetails>() {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/model/ImageDetails.java]: This method should be on top of class

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java [hunk 3]
```diff
@@ -16,6 +19,14 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
         ImagesCompoundRecyclerView imagesCompoundRecyclerView =
                 (ImagesCompoundRecyclerView) findViewById(R.id.images_compound_recycler_view);
-        new ImagesViewModel(imagesCompoundRecyclerView, getContentResolver());
+        new ImagesViewModel(imagesCompoundRecyclerView,
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java]: I prefer to move parameter under the Initializer like:
new ImagesViewModel(
                    imagesCompoundRecyclerView, 
                    etc,

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesViewModel.java [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
 import com.invisibleteam.goinvisible.mvvm.images.adapter.ImagesCompoundRecyclerView;
 
 class ImagesViewModel {
     private ImagesCompoundRecyclerView imagesCompoundRecyclerView;
-    private ContentResolver contentResolver;
+    private ImagesProvider imagesProvider;
+    private ImagesView imagesView;
 
-    ImagesViewModel(ImagesCompoundRecyclerView imagesCompoundRecyclerView, ContentResolver contentResolver) {
+    ImagesViewModel(ImagesCompoundRecyclerView imagesCompoundRecyclerView,
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesViewModel.java:11]: I prefer to put arguments in a column

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesCompoundRecyclerView.java [hunk 5]
```diff
@@ -40,4 +40,8 @@ public void updateResults(List<ImageDetails> imageList) {
         imageItemAdapter.updateImageList(imageList);
         imageItemAdapter.notifyDataSetChanged();
     }
+
+    public void setOnItemClickListener(ImagesItemAdapter.OnClickListener onItemClickListener) {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesCompoundRecyclerView.java]: Hungarian name is redundant (onItemClickListener) listener is enough

**Review by BartSoftware — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesCompoundRecyclerView.java (hunk 5 above)
Comment (kejmil01) [reply app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesCompoundRecyclerView.java]: Done

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesViewModel.java (hunk 4 above)
Comment (kejmil01) [reply app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesViewModel.java:11]: Done

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java (hunk 3 above)
Comment (kejmil01) [reply app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java]: Done

File: app/src/main/java/com/invisibleteam/goinvisible/model/ImageDetails.java (hunk 2 above)
Comment (kejmil01) [reply app/src/main/java/com/invisibleteam/goinvisible/model/ImageDetails.java]: Done

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (kejmil01) [reply app/src/main/AndroidManifest.xml]: Done

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesItemAdapter.java [hunk 6]
```diff
@@ -39,6 +47,10 @@ void updateImageList(List<ImageDetails> imageList) {
         this.imageList = imageList;
     }
 
+    void setOnItemClickListener(OnItemClickListener onItemClickListener) {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesItemAdapter.java]: listener

**Review by BartSoftware — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesItemAdapter.java (hunk 6 above)
Comment (kejmil01) [reply app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/adapter/ImagesItemAdapter.java]: Done

**Review by BartSoftware — APPROVED** [review]

**Pull request merged**